hledger-lib/Hledger/Reports/AccountTransactionsReport.hs

```diff
@@ -94,63 +95,71 @@ triCommodityBalance c = filterMixedAmountByCommodity c  . triBalance
 accountTransactionsReport :: ReportSpec -> Journal -> Query -> AccountTransactionsReport
 accountTransactionsReport rspec@ReportSpec{_rsReportOpts=ropts} j thisacctq = items
   where
-    -- a depth limit should not affect the account transactions report
-    -- seems unnecessary for some reason XXX
-    reportq = simplifyQuery $ And [_rsQuery rspec, periodq, excludeforecastq (forecast_ ropts)]
+    -- a depth limit should not affect the account transactions report, nor
+    -- should cur or amt queries, as they are handled separately
```

Comment:
They are handled separately because they need to be matched before valuation. They are then removed from the query, because otherwise it tries to match again after valuation, which will mean any cur: queries will fail if they have been converted.

Reply:
Good. What do you think about
```
    -- A depth limit should not affect the account transactions report; it should show all transactions in/below this account.
    -- Queries on currency or amount are also ignored at this stage; they are handled earlier, before valuation.
```